app/controllers/api_controller.rb

```diff
@@ -0,0 +1,8 @@
+class ApiController < ApplicationController
+   skip_before_action :verify_authenticity_token
+   private
+   def authenticated?
+ # #2
+     authenticate_or_request_with_http_basic {|username, password| User.where( username: username, password: password).present? }
```

Comment:
Since this line is pretty long (well over 100 characters), I would usually break it up into a multiline block (`do` ...  `end` style instead of `{` ... `}`)